plugins/org.jboss.tools.livereload.core/src/org/jboss/tools/livereload/core/internal/util/WSTUtils.java

```diff
@@ -186,11 +193,39 @@ public static IServer extractServer(final String browserLocation, final List<ISe
 	}
 
 	/**
+	 * Retrieves the hostname for the given server, with support for specific types of servers, such as OpenShift 3.
+	 * @param server the server to analyze
+	 * @return the server hostname or <code>null</code> if it could not be determined 
+	 * @throws CoreException if something went wrong
+	 */
+	public static String getWebHost(final IServer server) throws CoreException {
+		final String serverType = server.getServerType().getId();
```

Comment:
I think andre's fixing it to use the actual host and use setHost() properly, so I'm not sure if this is needed.